### PR TITLE
have --open use the options.public path if it exists

### PR DESCRIPTION
**Pull request by bdwain**
This changes the behavior of the --open flag to open the browser to the --public url if it is specified. (the default will be the host like before).

The original issue is #739 , though it will also address #725 by letting you specify localhost as the --public url.

This is not a breaking change.

This is also addressed for version 1 with #748

### CODE REVIEW

File: bin/webpack-dev-server.js [hunk 1]
```diff
@@ -330,14 +330,14 @@ function startDevServer(wpOpt, options) {
...
 	// the formatted domain (url without path) of the webpack server
-	const domain = url.format({
+	const domain = options.public ? `${protocol}://${options.public}` : url.format({
 		protocol: protocol,
 		hostname: options.host,
 		port: options.socket ? 0 : options.port.toString()
 	});
 
 	if(options.inline !== false) {
-		const devClient = [`${require.resolve("../client/")}?${options.public ? `${protocol}://${options.public}` : domain}`];
```
Comment (bdwain) [inline bin/webpack-dev-server.js]: this line behaves the same, but domain is also used on line 375 to calculate the url that gets opened, and that's where the behavior changes

**Comment by bdwain** [comment]
@SpaceK33z Is there anything else you wanted done for this?

**Comment by SpaceK33z** [comment]
@bdwain, I am planning to merge this after the 2.2 release (2.2 has been published but not yet with the `latest` tag). So it's fine!

**Comment by bdwain** [comment]
cool!

**Comment by SpaceK33z** [comment]
Thanks!

**Pull request merged**